serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/BillingFixtures.scala

```diff
@@ -1,171 +1,80 @@
+    BillingFixtures
+      .temporaryBillingProject[IO](billingAccountName, creatorAuthToken, prefix, owners, users)
+      .use(projectName => IO.delay(testCode(projectName)))
+      .unsafeRunSync
 
   /**
-   * Manually claim a project provisioned by GPAlloc and optionall add members.
-   * As opposed to `Credentials`, accepts `AuthToken` and `String` values for the new owner.
-   * This way a GPAlloc project can be claimed as a pet SA.
+   * Create a v2 billing project `Resource` whose lifetime is bound to the `Resource`'s `use` method.
    *
-   * @param newOwnerEmail Email for the new billing project owner
-   * @param ownerEmails a List of emails (as Strings) to add as owners of this project
-   * @param userEmails a List of emails (as Strings) to add as users of this project
-   * @param newOwnerToken Function that returns an AuthToken for the new billing project owner to pass to billing project endpoints
-   * @return Some(GPAllocProject) if it succeeded, none if it failed
+   * @param billingAccountName Name of Google billing account the new billing project will bill to.
+   * @param creatorAuthToken   Auth token of billing project creator
+   * @param prefix             Prefix for billing project name.                  [default: "tmp-billing-project-"]
+   * @param owners             Additional billing project owner email addresses. [default: None]
+   * @param users              Additional billing project user email addresses   [default: None]
+   * @return
    */
-  def claimGPAllocProject(newOwnerEmail: String, ownerEmails: List[String], userEmails: List[String])(
-    newOwnerToken: () => AuthToken
-  ): ClaimedProject =
-    // request a GPAlloced project as the potential new owner
-    GPAlloc.projects.requestProject(newOwnerToken()) match {
-      case Some(project) =>
-        // the Rawls endpoint to register a precreated project needs to be called by a Rawls admin
-        // but it also takes the new owner's UserInfo in order to create the resource as them in Sam
-        val adminToken = UserPool.chooseAdmin.makeAuthToken()
-        val newOwnerUserInfo =
-          UserInfo(OAuth2BearerToken(newOwnerToken().value), WorkbenchUserId("0"), WorkbenchEmail(newOwnerEmail), 3600)
-        Rawls.admin.claimProject(project.projectName, project.cromwellAuthBucketUrl, newOwnerUserInfo)(adminToken)
-
-        addMembersToBillingProject(project.projectName, ownerEmails, BillingProjectRole.Owner)(newOwnerToken())
-        addMembersToBillingProject(project.projectName, userEmails, BillingProjectRole.User)(newOwnerToken())
-
-        ClaimedProject(project.projectName, gpAlloced = true)
-      case _ =>
-        throw new Exception("claimGPAllocProject got no project back from GPAlloc")
+  def temporaryBillingProject[F[_]](billingAccountName: String,
+                                    creatorAuthToken: AuthToken,
+                                    prefix: Option[String] = None,
```

Comment:
I like the idea of being explicit up front and using a default parameter value, but it's not a strong preference.
```suggestion
                                    prefix: String = "tmp-",
```

Reply:
I spent a long time thinking about this and I went with `Option` because:
1. In the signature it tells me which parameters are optional
3. it allows me to know if a user has specified preference or not and compute defaults from other arguments. 

Consider the following example. Let's say I want to default the clone billing project to the source billing project and compute a prefix for the clone based upon the name of the source workspace. I cannot do this using default parameter values because default parameter values cannot be dependent on actual parameter values.
```scala
def temporaryWorkspaceClone(
  sourceBillingProject: String,
  sourceWorkspaceName: String.
  cloneBillingProject: String = sourceBillingProject,
  cloneWorkpacePrefix: String = sourceWorkspaceName + "-clone-",
   ...
```
For this reason, using `Option` provides a consistent interface for documenting and handling default parameters.

Reply:
love this discussion!  If we ever wanted to start creating a team style guide, this is the sort of thing I would put there.